.pre-commit-config.yaml

```diff
@@ -1,21 +1,31 @@
 repos:
   - repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v4.5.0
+    rev: v5.0.0
     hooks:
       - id: check-added-large-files
       - id: check-yaml
       - id: check-merge-conflict
       - id: end-of-file-fixer
 
-  - repo: local
+  - repo: https://github.com/astral-sh/uv-pre-commit
+    # uv version.
+    rev: 0.6.14
     hooks:
+      - id: uv-lock
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    # Ruff version.
+    rev: v0.11.4
+    hooks:
+      # Run the linter.
       - id: ruff
         name: lint with ruff
         language: system
-        entry: ruff check --force-exclude --fix
+        entry: ruff check --force-exclude
```

Comment:
Is this intentional? I find it useful that pre-commit makes the fixes for you.

Reply:
As a matter of fact I really don't like it/was planning to raise an issue about it, but I don't think the change should go in this PR either way.

Reply:
@callumforrester you want to remove pre-commit fixing?

Reply:
Not in this PR, no. 

I'm not a big fan of it because I forget it's there, all of our other automated checks don't have side effects so I keep forgetting this one does . So I commit, get an error message from the pre-commit, run `ruff check --fix` and it says "no problems found". 

What are other people's workflows? Commit, see error, notice the word "fixed", commit again, all happy?

Reply:
I see what you mean, that could be confusing. For me, in lazygit I run commit, get the error and see new unstaged changes appear on the files, so it is really obvious what has happened. But, that won't be the case for everyone. If the prompt at least had an unstaged lines counter that might help.